R/utils.R

```diff
@@ -0,0 +1,97 @@
+read_meta <- function(path, filename) {
+  if (meta_exists(path, filename)) {
+    utils::read.csv(file.path(path, "meta", filename),
+           stringsAsFactors = FALSE)
+  } else {
+    NULL
+  }
+}
+
+meta_exists <- function(path, filename) {
+  file.exists(file.path(path, "meta", filename))
+}
+
+data_frame <- function(...) {
+  data.frame(stringsAsFactors = FALSE, ...)
+}
+
+sql_in_char <- function(strings) {
+  paste0("('", paste(strings, collapse = "','"), "')")
+}
+
+sql_in_numeric <- function(numerics) {
+  paste0("(", paste(numerics, collapse = ","), ")")
+}
+
+sql_in <- function(things) {
+  if (is.character(things)) sql_in_char(things)
+  else if (is.numeric(things)) sql_in_numeric(things)
+  else stop("Can't convert things with sql_in")
+}
+
+db_get <- function(con, table, id_field = NULL, id_values = NULL, select = "*") {
+  sql <- sprintf("SELECT %s FROM %s", select, table)
+  if (!is.null(id_field)) {
+    sql <- sprintf("%s WHERE %s IN %s", sql, id_field, sql_in(id_values))
+  }
+  DBI::dbGetQuery(con, sql)
+}
+
+# Return a vector of characters for a table, each entry being all the fields
+# of that table mashed together, separated by '#'
+
+mash <- function(tab) {
+  fields <- sort(names(tab))
+  df_args <- c(tab, sep = "#")
```

Comment:
This is a bit of a worry, but I think `\r` is a better mash character if you do want to go this route